src/wallet/wallet.cpp

```diff
@@ -2063,15 +2062,15 @@ bool CWallet::AvailableCoins(std::vector<COutput>* pCoins,      // --> populates
 
                 // --Skip P2CS outputs
                 // skip cold coins
-                if (mine == ISMINE_COLD && (!fIncludeColdStaking || !HasDelegator(pcoin->vout[i]))) continue;
+                if (mine == ISMINE_COLD && ((coinControl && !coinControl->fIncludeColdStaking) || !HasDelegator(pcoin->vout[i]))) continue;
                 // skip delegated coins
                 if (mine == ISMINE_SPENDABLE_DELEGATED && !fIncludeDelegated) continue;
 
                 bool solvable = IsSolvable(*this, pcoin->vout[i].scriptPubKey);
 
                 bool spendable = ((mine & ISMINE_SPENDABLE) != ISMINE_NO) ||
                         (((mine & ISMINE_WATCH_ONLY) != ISMINE_NO) && (coinControl && coinControl->fAllowWatchOnly && solvable)) ||
-                        ((mine & ((fIncludeColdStaking ? ISMINE_COLD : ISMINE_NO) |
+                        ((mine & (coinControl && (coinControl->fIncludeColdStaking ? ISMINE_COLD : ISMINE_NO) |
```

Comment:
(d8a8ed49f8cb4f48577466a88e93e5ff5639f054)
This gives false `spendable` coins for watchonly.
Should either use the local `fIncludeColdStaking` suggested above, or move it (negated in OR) inside the parenthesis:
```cpp
((mine & ((!coinControl || coinControl->fIncludeColdStaking ? ISMINE_COLD : ISMINE_NO) |
```

This bug can be triggered, for example, running the `wallet_basic` functional test.